libsolidity/ast/AST.h

```diff
@@ -587,10 +587,16 @@ class ContractDefinition: public Declaration, public StructurallyDocumented, pub
 class IdentifierPath: public ASTNode
 {
 public:
-	IdentifierPath(int64_t _id, SourceLocation const& _location, std::vector<ASTString> _path):
-		ASTNode(_id, _location), m_path(std::move(_path)) {}
+	IdentifierPath(
+		int64_t _id,
+		SourceLocation const& _location,
+		std::vector<ASTString> _path,
+		std::vector<SourceLocation> _pathLocations
+	):
+		ASTNode(_id, _location), m_path(std::move(_path)), m_pathLocations(std::move(_pathLocations)) {}
```

Comment:
```suggestion
		ASTNode(_id, _location), m_path(std::move(_path)), m_pathLocations(std::move(_pathLocations))
		{
		    solAssert(m_pathLocations.size() == m_path.size(), "");
		}
```
I think they must always match, right?

Reply:
Depends on how we want to do the json import..